android-components/components/feature/search/src/main/java/mozilla/components/feature/search/storage/SearchEngineReader.kt

```diff
@@ -22,6 +23,19 @@ internal const val URL_TYPE_SUGGEST_JSON = "application/x-suggestions+json"
 internal const val URL_TYPE_SEARCH_HTML = "text/html"
 internal const val URL_REL_MOBILE = "mobile"
 internal const val IMAGE_URI_PREFIX = "data:image/png;base64,"
+internal const val GOOGLE_ID = "google"
+
+// List of general search engine ids, taken from
+// https://searchfox.org/mozilla-central/rev/ef0aa879e94534ffd067a3748d034540a9fc10b0/toolkit/components/search/SearchUtils.sys.mjs#200
+internal val GENERAL_SEARCH_ENGINE_IDS = setOf(
+    GOOGLE_ID,
```

Comment:
I think we could've also just put `google-b-m` and `google-com-nocodes` instead of doing a `startsWith` as well. No strong opinions here. I went with `startsWith` because I saw there was `google-b-amzes` for Echo Show in my search https://searchfox.org/mozilla-mobile/search?q=google-&path=&case=false&regexp=false